speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp

```diff
@@ -213,6 +220,7 @@ static void SaveSettingsToDisk()
 /// @brief Saves the settings into the frontend settings and the backend config
 static void SaveSettings(void* /* dummy */)
 {
+    if (!m_blackBoxChosen) { return; }
```

Comment:
It's very confusing when you click apply and it doesn't work, it would be nice if there was some error message that you haven't selected a black box yet.

Reply:
I agree, the user needs some feedback on why they can't continue.